sdks/java/io/kafka/src/main/java/org/apache/beam/sdk/io/kafka/KafkaReadSchemaTransformProvider.java

```diff
@@ -178,16 +180,45 @@ public PCollectionRowTuple expand(PCollectionRowTuple input) {
       if (confluentSchemaRegUrl != null) {
         final String confluentSchemaRegSubject =
             checkArgumentNotNull(configuration.getConfluentSchemaRegistrySubject());
-        KafkaIO.Read<byte[], GenericRecord> kafkaRead =
-            KafkaIO.<byte[], GenericRecord>read()
-                .withTopic(configuration.getTopic())
-                .withConsumerFactoryFn(new ConsumerFactoryWithGcsTrustStores())
-                .withBootstrapServers(configuration.getBootstrapServers())
-                .withConsumerConfigUpdates(consumerConfigs)
-                .withKeyDeserializer(ByteArrayDeserializer.class)
-                .withValueDeserializer(
-                    ConfluentSchemaRegistryDeserializerProvider.of(
-                        confluentSchemaRegUrl, confluentSchemaRegSubject));
+        KafkaIO.Read<byte[], GenericRecord> kafkaRead;
+
+        if (confluentSchemaRegUrl.contains(googleManagedSchemaRegistryPrefix)) {
```

Comment:
For readability we may want to generalize this logic. For example, put common with... out of if clause, and create a helper method to handle certain confluentSchemaRegUrl. This makes it easy to extend in the future